src/switch_snmp.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python
+#
+# igcollect - Switch port traffic over SNMP
+#
+# Copyright (c) 2016, InnoGames GmbH
+#
+
+from __future__ import print_function
+from argparse import ArgumentParser
+
+import re
+import subprocess
+import time
+
+port_name_oid = '.1.3.6.1.2.1.31.1.1.1.1'
+procurve_trunk_oid = '.1.3.6.1.4.1.11.2.14.11.5.1.7.1.3.1.1.8'
+cisco_agg_oid = '.1.2.840.10006.300.43.1.2.1.1.12'
+
+COUNTERS = {
+    'bytesIn': '.1.3.6.1.2.1.31.1.1.1.6',
+    'bytesOut': '.1.3.6.1.2.1.31.1.1.1.10',
+    'pktsIn': '.1.3.6.1.2.1.31.1.1.1.7',
+    'pktsOut': '.1.3.6.1.2.1.31.1.1.1.11',
+    'brdPktsIn': '.1.3.6.1.2.1.31.1.1.1.9',
+    'brdPktsOut': '.1.3.6.1.2.1.31.1.1.1.13',
+    'ifInErrors': '.1.3.6.1.2.1.2.2.1.14',
+    'ifOutErrors': '.1.3.6.1.2.1.2.2.1.20',
+    'ifInDiscards': '.1.3.6.1.2.1.2.2.1.13',
+    'ifOutDiscards': '.1.3.6.1.2.1.2.2.1.19',
+}
+
+
+def get_snmp_value(host, community, node_name):
+    ''' Function retriving value of given command.
+        In the current version only one row response is handled:
+        return last value retireved via SNMP
+    '''
+
+    p = subprocess.Popen(
+        ['snmpwalk', '-On', '-c', community, '-v2c', host, node_name],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
```

Comment:
Exit code of this is not checked.  Why don't you just use `subprocess.check_output()`?